pytest/tests/test_sudo.py

```diff
@@ -0,0 +1,634 @@
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.BareAD)
+@pytest.mark.topology(KnownTopology.BareLDAP)
+def test_sudo__duplicate_sudo_user(client: Client, provider: GenericProvider):
+    """
+    :title: User is mentioned twice in sudoUser attribute, once with shortname and once with fully qualified name
+    :setup:
+        1. Create users "user-1", "user-2", "user-3", "user-4"
+        3. Create sudorule to allow "user-1", "user-2", "user-2@test", "user-3" run "/bin/ls on all hosts
+        4. Enable SSSD sudo responder
+        5. Start SSSD
+    :steps:
+        1. List sudo rules for "user-1", "user-2", "user-3"
+        2. Run "sudo /bin/ls root" as "user-1", "user-2", "user-3"
+        3. List sudo rules for "user-4"
+        4. Run "sudo /bin/ls root" as "user-4"
+    :expectedresults:
+        1. User is able to run /bin/ls as root
+        2. Command is successful
+        3. User is not able to run /bin/ls as root
+        4. Command failed
+    :customerscenario: False
+
+    Note: This test can not run on IPA since it will not allow this case to happen.
+    """
+    provider.user("user-1").add()
+    provider.user("user-2").add()
+    provider.user("user-3").add()
+    provider.user("user-4").add()
+    provider.sudorule("test").add(
+        user=["user-1", "user-2", f"user-2@{client.sssd.default_domain}", "user-3"], host="ALL", command="/bin/ls"
+    )
+
+    client.sssd.common.sudo()
+    client.sssd.start()
+
+    # Try several users to make sure we don't mangle the list
+    for user in ["user-1", "user-2", "user-3"]:
+        assert client.auth.sudo.list(user, "Secret123", expected=["(root) /bin/ls"]), "Sudo list failed!"
+        assert client.auth.sudo.run(user, "Secret123", command="/bin/ls /root"), "sudo command failed!"
+
+    assert not client.auth.sudo.list("user-4", "Secret123"), "Sudo list successful!"
+    assert not client.auth.sudo.run("user-4", "Secret123", command="/bin/ls /root"), "Sudo command successful!"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.ticket(bz=1380436, gh=4236)
+@pytest.mark.topology(KnownTopology.BareAD)
+@pytest.mark.topology(KnownTopology.BareIPA)
+@pytest.mark.topology(KnownTopology.BareLDAP)
+def test_sudo__case_sensitive_false(client: Client, provider: GenericProvider):
+    """
+    :title: Sudo rules work correctly for case-insensitive domains
+    :setup:
+        1. Create user "USER-1"
+        2. Create sudorule to allow "user-1" run "/bin/less on all hosts
+        3. Create sudorule to allow "USER-1" run "/bin/more on all hosts
+        4. Enable SSSD sudo responder
+        5. Set "case_sensitive" to "false"
+        6. Start SSSD
+    :steps:
+        1. List sudo rules for "user-1"
+        2. Run "sudo /bin/less root" as user-1
+        3. Run "sudo /bin/more root" as user-1
+        4. List sudo rules for "USER-1"
+        5. Run "sudo /bin/less root" as USER-1
+        6. Run "sudo /bin/more root" as USER-1
```

Comment:
I think, the commands should be  
"sudo /bin/less /root/test"
 "sudo /bin/more /root/test"
Or
Is it different

Reply:
done